public/docs/ts/latest/guide/dependency-injection.jade

```diff
@@ -748,105 +717,117 @@ code-example(format, language="html").
-- var lang = current.path[1]
-- var opaquetoken = lang == 'dart' ? '<code>OpaqueToken</code>' : '<a href="../api/core/OpaqueToken-class.html"><code>OpaqueToken</code></a>'
-h3 OpaqueToken
-p.
   The solution is to define and use an !{opaquetoken}.
   The definition looks like this:
-// #enddocregion tokens-opaque-1
+
 +makeExample('dependency-injection/ts/app/app.config.ts','token')(format='.')
+
 :marked
   We register the dependency provider using the `OpaqueToken` object:
-+makeExample('dependency-injection/ts/app/providers.component.ts','providers-9b')(format=".")
-// #docregion tokens-opaque-2
-- var lang = current.path[1]
-- var decorated = lang == 'dart' ? 'annotated' : 'decorated'
-- var configuration = lang == 'dart' ? '' : 'configuration'
-:marked
-  Now we can inject the #{configuration} object into any constructor that needs it, with
-  the help of an `@Inject` #{decorator} that tells Angular how to find the #{configuration} dependency value.
-// #enddocregion tokens-opaque-2
-+makeExample('dependency-injection/ts/app/providers.component.ts','provider-9b-ctor')(format=".")
-
-// begin Typescript only
+
++makeExample('dependency-injection/ts/app/providers.component.ts','providers-9')(format=".")
+
+:marked
+  Now we can inject the configuration object into any constructor that needs it, with
+  the help of an `@Inject` #{_decorator}:
+
++makeExample('dependency-injection/ts/app/app.component.2.ts','ctor')(format=".")
+
+- var configType = _docsFor == 'dart' ? '<code>Map</code>' : '<code>AppConfig</code>'
 .l-sub-section
   :marked
     Although it plays no role in dependency injection,
-    the `Config` interface supports strong typing of the configuration object within the class.
+    the !{configType} interface supports typing of the configuration object within the class.
+
+block dart-map-alternative
+  :marked
+    Or we can provide and inject the configuration object in our top-level `AppComponent`.
+
+  +makeExcerpt('app/app.component.ts','providers')
+
+#optional
 :marked
-// end typescript only
+  ## Optional dependencies
+
+  Our `HeroService` *requires* a `Logger`, but what if it could get by without
+  a logger?
+  As is illustrated next, we can indicate that to the dependency injection
+  framework by marking the constructor's argument as optional.
+
++ifDocsFor('ts')
+  +makeExample('dependency-injection/ts/app/providers.component.ts','import-optional', '')
++makeExample('dependency-injection/ts/app/providers.component.ts','provider-10-ctor', '')(format='.')
 
-// Skip for Dart (we have another example)
 :marked
-  Or we can provide and inject the configuration object in our top-level `AppComponent`.
-+makeExample('dependency-injection/ts/app/app.component.ts','providers', 'app/app.component.ts (providers)')(format=".")
-+makeExample('dependency-injection/ts/app/app.component.ts','ctor', 'app/app.component.ts (constructor)')(format=".")
+  Be prepared for a null logger. If we don't register one somewhere up the line,
```

Comment:
Instead of 2nd person here, let's continue with a "we" focus. Maybe:

When using `@Optional()`, our code must be prepared for a null value. If we don't register a logger somewhere up the line, the injector will set the value of `logger` to `null`.